lib/asset/transferpart.js

```diff
@@ -34,13 +35,13 @@ class TransferPart {
      */
     constructor(transferAsset, targetUrls, contentRange, targetHeaders) {
         if (!(transferAsset instanceof TransferAsset)) {
-            throw Error(`Invalid transfer asset: ${transferAsset}`);
+            throw new IllegalArgumentError("Invalid transfer asset", transferAsset);
```

Comment:
From the text we wouldn't know why it's invalid. You described why in other errors (wrong type), so why not here too?